fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -164,9 +187,20 @@ where
             Err(e) => {
                 let err = e.to_string();
                 if is_null_round_str(&err) {
-                    tracing::debug!(height, "detected null round at height");
-
-                    self.sync_pointers.advance_head();
+                    tracing::debug!(
+                        height,
+                        "detected null round at height, inserted None to cache"
+                    );
+
+                    atomically_or_err::<_, Error, _>(|| {
+                        self.provider.new_parent_view(height, None)?;
+                        emit!(EventType::NewParentView, is_null = true, height);
```

Comment:
It's okay that we do debug logs inside `atomically`, but I would try not to `emit!` any events because these blocks can be retried on conflict, so any side effecting should be avoided. I think we should treat `emit!` a bit more seriously than throwaway debug logs, and try to make it consistent; emitting the same event twice in a row would be weird, and could throw off metrics too.